### DISCUSSION TITLE
Extending the plugin to handle two separate push providers

### DISCUSSION THREAD

**Discussion by jacquesdev** [invertase/react-native-firebase#5619] [Ideas]
I'd like to use this plugin in my app to handle push notifications from two separate push providers, but am not currently able to do that (as far as I can see). If this is not currently possible in this module, is it possible to add something like this? 

It looks like we need the ability in react-native to somehow override methods onNewToken and onMessageReceived?

**Comment by mikehardy** [comment]
Hi there!
"this plugin" --> which plugin exactly? on what platforms? I assume messaging, cross platform

"two separate push providers" --> I'm not sure how this is relevant? There is actually only one way to receive cloud messages I'm aware of at the device level, FCM on Android, APNs on iOS. This module here on device is more of a "receiver", not a "pusher", Push away, from OneSignal, Firebase, direct to google or Apple REST APIs, however you like.

The tokens and message receipt here are the FCM token and APNs tokens, and receipt of FCM and APNs messages, if there are other actual cloud-to-device message send styles that we're missing, then they would need addition at the native level I think.

Perhaps you are looking for Huawei or similar support?

With a great deal more specifics, I can perhaps think about it in a less vague way.

**Reply by jacquesdev** [reply]
Hello Mike

Apologies for not being more detailed. As to your first question, yes, I do mean the mesaging plugin.

As to the second question, it's the ability for the messaging plugin to handle (received) push notifications from two "pushers" (for example OneSignal and Firebase direct). 

In my specific use case I'm currently using Firebase direct for my own backend to send push notifications (which are currently handled by your messaging plugin) and trying to integrate with another provider who happens to be using Firebase as well. 

In summary I need one single receiver plugin (the messaging plugin), to have the ability to receive push notifications from two "pushers".

Hopefully that clears things up, please let me know if you have any other questions.

**Reply by mikehardy** [reply]
So you fall in this part of my speculative text:

> There is actually only one way to receive cloud messages I'm aware of at the device level, FCM on Android, APNs on iOS. This module here on device is more of a "receiver", not a "pusher", Push away, from OneSignal, Firebase, direct to google or Apple REST APIs, however you like.

It should work fine.

What did your testing indicate?
